### ISSUE TITLE
GetOrCreateCompilationWithAnalyzersAsync throwing null reference exception in integration tests

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn#62381]
Looking at integration test runs I'm seeing this stack in one of the log files (ManagedLanguage.IDE.RemoteHostClient). I'm not sure why this isn't being reported or failing tests. Looks like one of the !s on https://github.com/dotnet/roslyn/blob/e8dbd4281676b3fc20999b1f94f57abfac15c228/src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.CompilationManager.cs#L31 may not be correct?

```
07/01/2022 19:29:29 Coordinated Universal Time: Error : 1 :[devenv:8792] Unexpected exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.CodeAnalysis.Diagnostics.EngineV2.DiagnosticIncrementalAnalyzer.<GetOrCreateCompilationWithAnalyzersAsync>d__3.MoveNext() in /_/src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.CompilationManager.cs:line 30
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.CodeAnalysis.Diagnostics.EngineV2.DiagnosticIncrementalAnalyzer.<AnalyzeDocumentForKindAsync>d__88.MoveNext() in /_/src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs:line 54
```

**Comment by jasonmalinowski** [comment]
Turns out this was actually broken by #62257 -- that change was incorrectly causing us to remove the solution-level analyzers, which then exposed a bug in GetOrCreateCompilationWithAnalyzersAsync which is incorrectly assuming a compilation will always exist in the map. That's not correct if the project didn't have any analyzers -- we avoid creating a CompilationWithAnalyzers in that case.
